torch/_functorch/_aot_autograd/autograd_cache.py

```diff
@@ -219,6 +219,19 @@ def check_cacheable(gm: torch.fx.GraphModule):
         check_node_safe(node)
 
 
+def check_metadata_cacheable(metadata: ViewAndMutationMeta):
+    """
+    When view replay is turned on, we bypass autograd cache if
+    the output is aliased.
+    """
+    if config.view_replay_for_aliased_outputs:
+        for info in metadata.output_info:
+            if info.functional_tensor is not None:
+                raise BypassAOTAutogradCache(
+                    "Cannot cache a graph with functional tensor"
```

Comment:
minor nit: the error message could be a bit more clear "We don't currently support caching on graphs that need to support view-replay" (maybe something about how this applies to graphs that return views of inputs or intermediates)